File: python/ImplementStrStr.py
__author__ = ' Zhen Wang'

# 28. Implement strStr()
class ImplementStrStr:
    def strStr(self, haystack: str, needle: str) -> int:
        if needle == "":
            return 0
        lenH = len(haystack)
        lenN = len(needle)
        for i in range(lenH - lenN + 1):
            if haystack[i: i + lenN] == needle:
                return i
        return -1

File: src/cc/ArraysAndStrings/ImplementStrStr.java
package cc.ArraysAndStrings;
//28. Implement strStr()
public class ImplementStrStr
{
    public int strStr(String haystack, String needle) {
        int len1 = haystack.length();
        int len2 = needle.length();
        for(int i = 0; i <= len1 - len2; i++){
            if(haystack.substring(i, i + len2).equals(needle)){
                return i;
            }
        }
        return -1;
    }
}
